core/src/main/scala/org/apache/spark/scheduler/TaskSetManager.scala

```diff
@@ -1033,8 +1037,8 @@ private[spark] class TaskSetManager(
         info.host, info.executorId, index, failureReason))
       numFailures(index) += 1
       if (numFailures(index) >= maxTaskFailures) {
-        logError("Task %d in stage %s failed %d times; aborting job".format(
-          index, taskSet.id, maxTaskFailures))
+        logError(log"Task ${MDC(TASK_ID, index)} in stage ${MDC(STAGE_ID, taskSet.id)} failed " +
```

Comment:
task id and task index are different things. Here it's task index. @gengliangwang

Reply:
You are right. I will find time to revisit all the usages of the `TASK_ID` log key.

Reply:
and TASK_ATTEMPT_ID should be the same as TASK_ID?

Reply:
@cloud-fan I created https://github.com/apache/spark/pull/46951
@pan3793 what do you mean by the same?

Reply:
https://github.com/apache/spark/blob/772d7d3aba17ca251b9464503963cc30635f69c0/core/src/main/scala/org/apache/spark/executor/Executor.scala#L639

Reply:
yes these two are the same